libraryBuilds/com.fasterxml.jackson.core/com.fasterxml.jackson.core-jackson-databind/src/test/java/com/github/j2objccontrib/j2objcgradle/issues/J2objcIssue639.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2015 the authors of j2objc-common-libs-e2e-test
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.github.j2objccontrib.j2objcgradle.issues;
```

Comment:
Good to have this test, it shouldn't be tied to a specific issue since this covers a number of different tests.

Rename and move to:
j2objc-common-libs-e2e-test/libraryBuilds/com.fasterxml.jackson.core/com.fasterxml.jackson.core-jackson-databind/src/test/java/com/github/j2objccontrib/CoreJacksonDataBind.java

Or use whatever filename you think is appropriate for the tests that it covers.

Then within the specific test that fails. Add a comment on it being a failure and then link to the issue:
https://github.com/google/j2objc/issues/639

Right now, when you just say "J2objcIssue639", it's unclear on which repository you're referring to. So a full link is always preferred.

Finally, issue #639 was closed already. Does that need to be reopened with more information? Either which way, keep the comments in there.